### ISSUE TITLE
ansible copy does not work with sudo

### ISSUE THREAD

**Issue by youhongshen** [ansible/ansible#12638]
Hi, I have a problem where the copy module does not work with sudo enabled.  And it breaks with a very specific combination of things.

My playbook:

```
  - name: copy 1
    copy: src=/tmp/joan/1 dest=/tmp/joan
    sudo: yes 

  - name: copy 2
    copy: src=/tmp/joan/2 dest=/tmp/joan
    sudo: yes
```

result:

```
 ______________
< TASK: copy 1 >
 --------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


ok: [icg1dbe01-dev]
 ______________
< TASK: copy 2 >
 --------------
        \   ^__^
         \  (oo)\_______
            (__)\       )\/\
                ||----w |
                ||     ||


fatal: [icg1dbe01-dev] => ssh connection error while waiting for sudo password prompt

FATAL: all hosts have already failed -- aborting
```

It's kinda weird that the first copy works but the 2nd doesn't, but that's not my concern right now.

Here's a combination of what works and what doesn't.

If I use "transport = paramiko" in the local machine (where local machine = the machine where ansible runs), it works.

If I take out the sudo: in the copy, it works

If the I turn off sudo requires password on the remote machine like this, it works.
 `%wheel  ALL=(ALL)       NOPASSWD: ALL`

If the remote machine has python = 2.6, it works.

```
      $ /usr/bin/python
      Python 2.6.6 (r266:84292, Oct 15 2013, 07:32:41)
```

If the remote machine has python 2.7, it doesn't work

```
     $ python
     Python 2.7.5 (default, May  8 2014, 17:35:19)
```

both my local and remote machine are running Oracle enterprise linux.
- local machine: oracle linux 7
- remote machine with python 2.6 is running oracle linux 6
- remote machine with python 2.7 is runinng oracle linux 7

to summarize: to reproduce this problem
- do not use paramiko for connection
- remote machine needs to require password for sudo
- remote machine's default python is on 2.7

ansible 1.9.0.1

I also tried with ansible 1.9.3 and 2.0 (w/ git pull --rebase as of earlier today) and both has the same problem.

**Comment by youhongshen** [comment]
sorry.. had a typo in the "how to reproduce the problem"

to summarize: to reproduce this problem
- do not use paramiko for connection
- remote machine needs to require password for sudo
- remote machine's default python is on ~~2.6~~ <--  **this should say 2.7**

**Comment by jctanner** [comment]
Hi!

Thanks very much for your submission to Ansible.  It sincerely means a lot to us. 

Testing this on the version you have specified, we cannot reproduce this problem.

Our reasons and evidence for this are:
- Tested ansible 2.1 from a fedora23 control host to an ubuntu14.04 host (python 2.7.6) with a remote user who required a sudo password. No errors occured in either of the copy tasks.

And we suggest that:
- Retest with a more recent version of ansible and if necessary file a new bug with full -vvvv output.

As such, we are most likely going to close this ticket.  Should you disagree with this conclusion, you may also wish to join our mailing lists, which are very active:
- https://groups.google.com/forum/#!forum/ansible-project - for user questions, tips, and tricks
- https://groups.google.com/forum/#!forum/ansible-devel - for strategy, future planning, and questions about writing code

Thank you once again for this and your interest in Ansible!
